packages/flutter_tools/lib/src/localizations/localizations_utils.dart

```diff
@@ -319,89 +321,112 @@ String generateReturnExpr(List<String> expressions, { bool isSingleStringVar = f
 
 /// Typed configuration from the localizations config file.
 class LocalizationOptions {
-  const LocalizationOptions({
-    this.arbDirectory,
-    this.templateArbFile,
-    this.outputLocalizationsFile,
+  LocalizationOptions({
+    String? arbDir,
+    this.outputDir,
+    String? templateArbFile,
+    String? outputLocalizationFile,
     this.untranslatedMessagesFile,
-    this.header,
-    this.outputClass,
-    this.outputDirectory,
+    String? outputClass,
     this.preferredSupportedLocales,
+    this.header,
     this.headerFile,
-    this.deferredLoading,
-    this.useSyntheticPackage = true,
-    this.areResourceAttributesRequired = false,
-    this.usesNullableGetter = true,
-    this.format = false,
-    this.useEscaping = false,
-    this.suppressWarnings = false,
-  });
+    bool? useDeferredLoading,
+    this.genInputsAndOutputsList,
+    bool? syntheticPackage,
+    this.projectDir,
+    bool? requiredResourceAttributes,
+    bool? nullableGetter,
+    bool? format,
+    bool? useEscaping,
+    bool? suppressWarnings,
+  }) : arbDir = arbDir ?? globals.fs.path.join('lib', 'l10n'),
```

Comment:
What are your thoughts on passing down `globals.fs` into `parseLocalizationsOptionsFromYAML` and `parseLocalizationsOptionsFromCommand` and call `fileSystem.path.join('lib', 'l10n')` at those two locations? I think `parseLocalizationsOptionsFromYAML` are used in several different locations, and I'd prefer to consolidate the default values if possible.

Reply:
To me hiding this fallback behavior deep in a helper function makes the overall behavior of whatever top level command that calls it more difficult to reason about, rather than lifting the fallback as high as possible and making the rest of the references to the value `final`.